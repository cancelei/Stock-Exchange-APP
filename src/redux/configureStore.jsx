import { configureStore } from '@reduxjs/toolkit';
import stocksReducers from './stocks/stockSlice';

export default configureStore({
  reducer: {
    stocks: stocksReducers,
  },
});
