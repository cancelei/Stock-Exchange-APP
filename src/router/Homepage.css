main {
  height: 200vh;
  background-color: #5788e4;
}
