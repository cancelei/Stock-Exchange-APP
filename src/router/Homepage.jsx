import './Homepage.css';
import { Col } from 'react-bootstrap';
import Navbar from '../components/NavElement/NavElement';
import DisplayStocks from '../components/displayData/displayStocks';

export default function Homepage() {
  return (
    <main className="mi-contenedor-con-max-width">
      <Col className="justify-content-center">
        <Navbar />
        <DisplayStocks />
      </Col>
    </main>
  );
}
