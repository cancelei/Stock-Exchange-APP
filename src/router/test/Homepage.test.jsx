import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../../redux/configureStore';
import '@testing-library/jest-dom/extend-expect';
import Homepage from '../Homepage';

describe('Tests the home page', () => {
  const component = (
    <Provider store={configureStore}>
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>
    </Provider>
  );

  test('Match the snapshot', () => {
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  test('Should contain an image component', () => {
    render(component);
    const main = screen.getByRole('main');
    expect(main).toBeInTheDocument();
  });
});
