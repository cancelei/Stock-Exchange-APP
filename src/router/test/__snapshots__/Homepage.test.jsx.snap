// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the home page Match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mi-contenedor-con-max-width"
      >
        <div
          class="justify-content-center col"
        >
          <header
            class="header"
          >
            <div
              bordered="true"
              class="d-flex justify-content-center container"
            >
              <div
                class="d-flex justify-content-center col"
              >
                <nav
                  class="navInner"
                >
                  <a
                    class="d-flex justify-content-center"
                    href="/"
                    style="margin: auto;"
                  >
                    <button
                      class="button-back"
                      type="submit"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
                        />
                      </svg>
                    </button>
                  </a>
                  <h2>
                    Stock Prices
                  </h2>
                  <h2>
                    by NYSE
                  </h2>
                </nav>
              </div>
            </div>
          </header>
          <div
            class="container"
          >
            <div
              class="d-flex justify-content-center mb-3"
            >
              <input
                placeholder="search stock"
                type="text"
                value=""
              />
            </div>
            <div
              class="row"
            >
              <div
                class="p-0 col"
              >
                <ul
                  class="list-group"
                />
              </div>
              <div
                class="p-0 col"
              >
                <ul
                  class="list-group"
                />
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mi-contenedor-con-max-width"
    >
      <div
        class="justify-content-center col"
      >
        <header
          class="header"
        >
          <div
            bordered="true"
            class="d-flex justify-content-center container"
          >
            <div
              class="d-flex justify-content-center col"
            >
              <nav
                class="navInner"
              >
                <a
                  class="d-flex justify-content-center"
                  href="/"
                  style="margin: auto;"
                >
                  <button
                    class="button-back"
                    type="submit"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
                      />
                    </svg>
                  </button>
                </a>
                <h2>
                  Stock Prices
                </h2>
                <h2>
                  by NYSE
                </h2>
              </nav>
            </div>
          </div>
        </header>
        <div
          class="container"
        >
          <div
            class="d-flex justify-content-center mb-3"
          >
            <input
              placeholder="search stock"
              type="text"
              value=""
            />
          </div>
          <div
            class="row"
          >
            <div
              class="p-0 col"
            >
              <ul
                class="list-group"
              />
            </div>
            <div
              class="p-0 col"
            >
              <ul
                class="list-group"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
